pkgs/games/shticker-book-unwritten/default.nix

```diff
@@ -0,0 +1,56 @@
+{ alsaLib, buildFHSUserEnv, fetchFromGitHub, lib, libX11, libXext, libglvnd, openssl, pkg-config, rustPlatform, stdenvNoCC }:
+let
+
+  pname = "shticker-book-unwritten";
+  version = "1.0.3";
+
+  shticker-book-unwritten-unwrapped = rustPlatform.buildRustPackage rec {
+    inherit pname version;
+
+    src = fetchFromGitHub {
+      owner = "JonathanHelianthicusDoe";
+      repo = "shticker_book_unwritten";
+      rev = "v${version}";
+      sha256 = "08lyxica0b0vvivybsvzigy2j7saar78mbz723y3g5hqrilfb5np";
+    };
+
+    cargoPatches = [ ./cargo-lock.patch ];
+    cargoSha256 = "1lnhdr8mri1ns9lxj6aks4vs2v4fvg7mcriwzwj78inpi1l0xqk5";
+
+    nativeBuildInputs = [ pkg-config ];
+
+    buildInputs = [ openssl ];
+  };
+
+  env = buildFHSUserEnv {
+    name = "${pname}-env-${version}";
+    targetPkgs = _: [
+      alsaLib
+      libX11
+      libXext
+      libglvnd
+      shticker-book-unwritten-unwrapped
+    ];
+    runScript = "shticker_book_unwritten";
+  };
+
+in stdenvNoCC.mkDerivation rec {
+  inherit pname version;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    ln -s ${env}/bin/* $out/bin/shticker_book_unwritten
```

Comment:
What does/should this line do?

Reply:
Links the buildFHSUserEnv-wrapped binary into the derivation output.

Reply:
I don't understand how that's supposed to work. Is `$out/bin/shticker_book_unwritten` supposed to be a directory? It might be a good idea to specify the destination directory using `-t`, e.g. `ln -s -t $out/bin ${env}/bin/*` or such, to make the intent more clear.

Reply:
Fair enough. I can see how it would seem like `$out/bin/shticker_book_unwritten` might be a directory. I was just copying what I saw another derivation (specifically `left4gore`) do. I will add the `-t` as you suggest to make it clearer

Reply:
ok, what exactly does it?... then I can help transforming it into something more readable

Reply:
I'm not sure I understand what the problem is. :confused:

The `buildFHSUserEnv` step effectively wraps the launcher in an FHS chroot, which is necessary because the game binary which it downloads and runs expects a typical FHS environment. The postInstall just links the wrapped binary from the `buildFHSUserEnv` step into the final derivation output.

Would `ln -s ${env}/bin/${pname}-env-${version} $out/bin/shticker_book_unwritten` be better?

Reply:
Ah. I can do `ln -s ${env}/bin/${env.name} $out/bin/shticker_book_unwritten`. I like this much better. :smile: What do you think?

Reply:
good.